app/models/manageiq/providers/ibm_power_hmc/infra_manager/event_target_parser.rb

```diff
@@ -20,35 +20,35 @@ def parse
     case ems_event.event_type
     when "MODIFY_URI", "ADD_URI", "DELETE_URI" # Damien: INVALID_URI?
       $ibm_power_hmc_log.info("#{self.class}##{__method__} #{ems_event.event_type} #{ems_event.full_data}")
-      elems = elem(raw_event[:data])
-      case elems[:type]
+      elems = raw_event[:data].match(/^.+\/rest\/api\/uom\/(ManagedSystem\/(?<manager_uuid>.+)\/)?(?<type>.+)\/(?<uuid>.+)$/).named_captures
```

Comment:
NOTE rubocop is warning about mixing captures and named captures, specifically
`(ManagedSystem\/(?<manager_uuid>.+)\/)*` but I need the entire group to be 0 or 1 of and don't actually need to capture the whole set.

Reply:
DO you actually need the outer group for capturing?  if not, then make it a non-capture group with `(?:ManagedSystem...)`

Reply:
> DO you actually need the outer group for capturing?

No

>  if not, then make it a non-capture group with 

Perfect, didn't know about this